openedx/core/djangoapps/import_export/courses/views.py

```diff
@@ -0,0 +1,530 @@
+                with the given name.  If there is no file in the directory with
+                the specified name, return None.
+                """
+                for fname, dirpath in get_all_files(directory):
+                    if fname == filename:
+                        return dirpath
+                return None
+
+            dirpath = get_dir_for_fname(course_dir, root_name)
+            if not dirpath:
+                self._save_request_status(request, courselike_string, -2)
+                return JsonResponse(
+                    {
+
+                        'error_message': _(
+                            'Could not find the {root_xml_file} file in the package.'
+                        ).format(root_xml_file=root_name),
+                        'stage': -2
+                    },
+                    status=415
+                )
+
+            dirpath = os.path.relpath(dirpath, data_root)
+            logging.debug('found %s at %s', root_name, dirpath)
+
+            log.info(
+                "Course import %s: Extracted file verified",
+                courselike_key
+            )
+            self._save_request_status(request, courselike_string, 3)
+
+            with dog_stats_api.timer(
+                'courselike_import.time',
+                tags=[u"courselike:{}".format(courselike_key)]
+            ):
+                courselike_items = import_func(
+                    modulestore(),
+                    request.user.id,
+                    settings.GITHUB_REPO_ROOT,
+                    [dirpath],
+                    load_error_modules=False,
+                    static_content_store=contentstore(),
+                    target_id=courselike_key,
+                )
+
+            new_location = courselike_items[0].location
+            logging.debug('new course at %s', new_location)
+
+            log.info(
+                "Course import %s: Course import successful", courselike_key
+            )
+            self._save_request_status(request, courselike_string, 4)
+
+        # Send errors to client with stage at which error occurred.
+        except Exception as exception:  # pylint: disable=broad-except
+            log.exception(
+                "error importing course"
+            )
+            return JsonResponse(
+                {
+                    'error_message': str(exception),
+                    'stage': -cache.get(status_key)
+                },
+                status=400
+            )
+
+        finally:
+            if course_dir.isdir():  # pylint: disable=no-value-for-parameter
+                shutil.rmtree(course_dir)
+                log.info(
+                    "Course import %s: Temp data cleared", courselike_key  # pylint: disable=no-value-for-parameter
+                )
+            # set failed stage number with negative sign in case of an
+            # unsuccessful import
+            if cache.get(status_key) != 4:
```

Comment:
So I realize you're only adapting existing code, but could you please assign these stage names some constants? It's low risk, and it's crazy to me that we actually labeled the steps 1, 2, 3, and 4 as opposed to giving them meaningful names.